src/Microsoft.TestPlatform.TestHostProvider/Hosting/DotnetTestHostManager.cs

```diff
@@ -565,6 +550,33 @@ bool SilentlyForceToX64()
         }
     }
 
+    internal /* for testing purposes */ void ForwardDotnetRootEnvironmentVariable(TestProcessStartInfo startInfo)
+    {
+        const string prefix = "VSTEST_WINAPPHOST_";
+        const string dotnetRoot = "DOTNET_ROOT";
+        string vstestDotnetRootEnvName = $"{prefix}{dotnetRoot}(x86)";
+
+        var vstestDotnetRootEnvValue = _environmentVariableHelper.GetEnvironmentVariable(vstestDotnetRootEnvName);
+        if (vstestDotnetRootEnvValue is null)
+        {
+            vstestDotnetRootEnvName = $"{prefix}{dotnetRoot}";
+            vstestDotnetRootEnvValue = _environmentVariableHelper.GetEnvironmentVariable(vstestDotnetRootEnvName);
+
+            if (vstestDotnetRootEnvValue is null)
+            {
+                EqtTrace.Verbose($"DotnetTestHostmanager.LaunchTestHostAsync: Prefix '{prefix}*' not found in env variables");
+                return;
+            }
+        }
+
+        string dotnetRootEnvName = Version.Parse(_targetFramework.Version) >= Version6_0
+            ? $"{dotnetRoot}_{_processHelper.GetCurrentProcessArchitecture().ToString().ToUpperInvariant()}"
```

Comment:
@MarcoRossignoli can you please remind me what does the "caller" do in this case? Is the `VSTEST_WINAPPHOST_..` set for .NET 6+ or not?

If it's set I find it weird that we intentionally ignore it here.

Reply:
Yes you are right @vitek-karas , I will add a check to ensure the variable is not set by user before this forward.

Thank you for the review!

Reply:
It's always set inside the SDK we don't know yet the tfm.